src/main/java/com/purbon/kafka/topology/roles/acls/AclsBindingsBuilder.java

```diff
@@ -111,10 +113,45 @@ public AclsBindingsBuilder(TopologyBuilderConfig config) {
     return bindingStream.map(TopologyAclBinding::new).collect(Collectors.toList());
   }
 
-  private Stream<AclBinding> producerAclsStream(String principal, String topic) {
-    return Stream.of(
-        buildTopicLevelAcl(principal, topic, PatternType.LITERAL, AclOperation.DESCRIBE),
-        buildTopicLevelAcl(principal, topic, PatternType.LITERAL, AclOperation.WRITE));
+  private Stream<AclBinding> producerAclsStream(Producer producer, String topic) {
+
+    List<AclBinding> bindings = new ArrayList<>();
+
+    bindings.addAll(
+        Arrays.asList(
+            buildTopicLevelAcl(
+                producer.getPrincipal(), topic, PatternType.LITERAL, AclOperation.DESCRIBE),
+            buildTopicLevelAcl(
+                producer.getPrincipal(), topic, PatternType.LITERAL, AclOperation.WRITE)));
+
+    producer
+        .getTransactionId()
+        .ifPresent(
+            transactionId -> {
+              bindings.add(
+                  buildTransactionIdLevelAcl(
```

Comment:
yes, true! however in some older versions this has given problems. This is why I keep adding DESCRIBE (_just in case TM_).

I will add an issue to remove it later on.